### DISCUSSION TITLE
Sharing on using Logica

### DISCUSSION THREAD

**Discussion by atqnp** [EvgSkv/logica#122] [Show and tell]
Hi all!

I wanted to share a little bit of my take on Logica. It is kind of like a tutorial and (for myself especially) as I explore its equivalent in basic SQL syntax. I am not a pro either in SQL or programming in general (still a newbie even in programming) so there may be mistakes that I may not recognize. I also just started writing actually😁. Feel free to comment or give any feedback if you find something!

https://atqnp.github.io/2021-05-27-logica-with-lego/

**Comment by EvgSkv** [comment]
Hi @atqnp !
Thanks for sharing! For people knowing SQL learning Logica by mapping their existing knowledge could indeed be natural.
The tutorial looks great! Here is a couple thoughts that came to my mind while reading:

1. Logica by now should have everything working on SQLite, we just want to try it out more before announcing full support.  
2. Logica's `Count(x)` corresponds to `APPROX_COUNT_DISTINCT(x)` function in BigQuery and `COUNT(DISTINCT x)` in Postgres, SQLite. Function SQL's `COUNT` counts rows with non-null values, but Logica's `Count`, as you said, counts distinct elements.

Just wanted to share even though these are minor :-)

Thanks again!

**Reply by atqnp** [reply]
Thank you very much for the feedback!
I will try to append my writing later.

**Comment by wowJasonCorderoy** [comment]
Thank you so much.
I am currently in the process of learning Logica and posts like this are life savers.
